frontend/packages/console-dynamic-plugin-sdk/scripts/generate-doc.ts

```diff
@@ -74,9 +75,29 @@ const renderDocNode = (docNode: tsdoc.DocNode): string => {
   return result;
 };
 
-const generateGitLinkDoc = (filePath: string) => ({
-  summary: `[For more details please refer the implementation](${filePath})`,
-});
+// Use typescript AST to traverse the VariableDeclaration initializer, find the first call to
+// require(), and resolve the import. If no call to require is found, returns null.
+const resolveRequire = (variableDeclaration: ts.VariableDeclaration) => {
+  return variableDeclaration.initializer?.forEachChild((node) => {
+    if (ts.isCallExpression(node) && node.expression.getText() === 'require') {
+      const [requireArgument] = node.arguments ?? [];
+      if (ts.isStringLiteral(requireArgument)) {
+        return require.resolve(requireArgument.text);
+      }
+    }
+    return null; // ts.Node.forEachChild keeps traversing until a truthy value is returned
+  });
+};
+
+const generateGitLinkDoc = (variableDeclaration: ts.VariableDeclaration) => {
+  const resolvedRequire = resolveRequire(variableDeclaration);
+  const absolutePath = resolvedRequire ?? variableDeclaration.getSourceFile().fileName;
+  const urlPath = absolutePath.replace(/.*((packages|public)\/.*)/, '$1');
+  const link = `${GITHUB_URL}/${urlPath}`;
```

Comment:
```suggestion
  const link = `../../../${urlPath}`;
```
Would this not always work? The docs are generated on a specific path (`frontend/packages/console-dynamic-plugin-sdk/docs/api.md`) -- couldn't we just work our way back to just before `packages|public` -- then let your `urlPath` build us up from there?

Reply:
This would only work if we were linking to other markdown files. I don't think GitHub Pages will let you link to any old file in the repo.

Reply:
"old file"? Everything is relative to the branch you're on. So if I am looking at a partial commit, all links that do not override will navigate me around in that commit. I'm fairly certain this is how it works. Have you seen actual issues or is this an assumption? Git is just a series of diffs on top of each other... the branch is just a pointer in the tree. I can't think of a reason this won't work 🤔 I have no experience with github pages, but I assume it works the same way as a repo.

Reply:
Yup this won't work on GitHub pages. You cannot link to a file 😒 Going to `master` will cause problems, almost guaranteed. Files move, get removed, change in a fashion that is not going to live well in documentation. Navigating to the latest is almost guaranteed to fail after a short period of time. 

I'm unsure if GitHub pages will allow us to link to the code 🤔 Maybe we should just start reporting errors to all things that do not have comments for them. Omit them from the readout and present errors that we have to fix before it publishes to our official docs. 

Can we instead cause a build failure somehow 🤔 If we crash the generate-doc when there is no documentation for something it may get horribly inconvenient to development cycles. Maybe write the failures out to a new file & write a check for it. We should be able to verify this on the CI as it builds and generates everything we have.

Reply:
I think this was initially implemented as a stop-gap until we could fill out all of the exposed APIs with TSDoc. 

That being said, I think it's okay that we fallback to a link to the API's source file if we don't have any TSDoc. IMO, that's better than omitting it from the docs entirely. The links should never be out of sync because we are using typescript compiler or require.resolve to dynamically build these paths.

Reply:
> "old file"?

Sorry, that was just an idiom 😆 "any-old-file"

Reply:
> ~That being said, I think it's okay that we fallback to a link to the API's source file if we don't have any TSDoc. IMO, that's better than omitting it from the docs entirely. The links should never be out of sync because we are using typescript compiler or require.resolve to dynamically build these paths.~

Hmm, okay, I think I'm getting what you mean now. Scratch this. I'm thinking in terms of what we are showing in GitHub Pages, but if someone is reading the markdown independently in a release branch, the links would be outdated. Maybe we just use a relative path without a hyperlink? Just say 'see the implementation at `frontend/packages/src/foo/bar.ts'.